src/auth/auth.ts

```diff
@@ -184,15 +193,15 @@ export type AuthTokenState = AuthTokenSuccessState | AuthTokenExpiredState | Aut
 export const loadAuthToken = async (
   args: OidcSignInArgs = { includeBillingScope: false, popUp: false }
 ): Promise<AuthTokenState> => {
-  const oldAuthTokenMetadata: TokenMetadata = authStore.get().authTokenMetadata;
-  const oldRefreshTokenMetadata: TokenMetadata = authStore.get().refreshTokenMetadata;
+  const oldAuthTokenMetadata: TokenMetadata = metricStore.get().authTokenMetadata;
+  const oldRefreshTokenMetadata: TokenMetadata = metricStore.get().refreshTokenMetadata;
 
   // for metrics, updates number of authToken load attempts for this session
-  authStore.update((state) => ({
+  metricStore.update((state) => ({
     ...state,
     authTokenMetadata: {
       ...oldAuthTokenMetadata,
-      totalTokenLoadAttemptsThisSession: authStore.get().authTokenMetadata.totalTokenLoadAttemptsThisSession + 1,
+      totalTokenLoadAttemptsThisSession: metricStore.get().authTokenMetadata.totalTokenLoadAttemptsThisSession + 1,
```

Comment:
The previous state of the store is available as the argument to the update function.

```suggestion
      totalTokenLoadAttemptsThisSession: state.authTokenMetadata.totalTokenLoadAttemptsThisSession + 1,
```